### PR TITLE
Add write protection for existing classes

### PR DESCRIPTION
**Pull request by geerteltink**
This PR adds detection if a class already exists. If it does exist it will throw an exception and won't overwrite the existing middleware.

### CODE REVIEW

File: src/CreateMiddleware/CreateMiddleware.php [hunk 1]
```diff
@@ -135,7 +139,6 @@ private function getComposerAutoloaders($projectRoot)
      */
     private function discoverNamespaceAndPath($class, array $autoloaders)
     {
-        $discoveredPath = false;
```
Comment (michalbundyra) [inline src/CreateMiddleware/CreateMiddleware.php:138]: It shouldn't be in this PR probably ;-)

File: src/GenerateProgrammaticPipelineFromConfig/Generator.php [hunk 2]
```diff
@@ -119,6 +119,7 @@ class Generator implements Constants
 
     /**
      * @param OutputInterface $console
+     * @param string $projectDir
```
Comment (michalbundyra) [inline src/GenerateProgrammaticPipelineFromConfig/Generator.php:122]: As above, shouldn't be in this PR.

File: src/Module/CommandCommonOptions.php [hunk 3]
```diff
@@ -19,6 +19,8 @@
 {
     /**
      * Add default arguments and options used by all commands.
+     *
+     * @param Command $command
```
Comment (michalbundyra) [inline src/Module/CommandCommonOptions.php:23]: As above. This and similar fixes are included in #30 ;-)

**Review by weierophinney — APPROVED** [review]

**Comment by weierophinney** [comment]
Thanks, @xtreamwayz!

**Pull request merged**